### ISSUE TITLE
Erroneous transaction (causes crash on click)

### ISSUE THREAD

**Issue by fabianrinker** [moneymanagerex/moneymanagerex#5926]
MMEX version:
 - [ ] 1.6.0
 - [ ] 1.6.1
 - [ ] 1.6.2
 - [x] 1.6.3
 - [ ] Other (please specify)

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [ ] Windows
 - [x] Mac OSX
 - [ ] Linux 

# Description of the bug
The following line with "Account error" and "Payee error" is visible:
![image](https://github.com/moneymanagerex/moneymanagerex/assets/13711459/12edcbac-1faf-4d8c-afac-75e477402b0b)

When double clicking the line, MMEX creates a debug report and crashes. Debug report was already sent by me and for a different reproduction one see at additional information.

# Reproduction

Is the bug reproducible? 
 - [x] Always (**Crash at double-click** on erroneous transaction)
 - [ ] Randomly
 - [x] Happened only once (probably only this **single erroneous line** was for some reason created currently in my db)

Reproduction steps:
Double-click on the line

Expected result:
open transaction, maybe showing more information about what is wrong.

Actual result:
Crash

# Additional information

```
<?xml version="1.0" encoding="UTF-8"?>
<report version="1.0" kind="exception">
  <system description="macOS Ventura Version 13.3.1 (a) (Build 22E772610a)"/>
  <stack>
    <frame level="0" function="wxDebugReport::AddContext(wxDebugReport::Context)" offset="0" address="0x10079c849"/>
    <frame level="1" function="wxDebugReport::AddAll(wxDebugReport::Context)" offset="0" address="0x10079b6ec"/>
    <frame level="2" function="mmGUIApp::ReportFatalException(wxDebugReport::Context)" offset="0" address="0x1003fc3b6"/>
    <frame level="3" function="wxFatalSignalHandler(int)" offset="0" address="0x100cdf7ab"/>
    <frame level="4" function="_sigtramp" offset="0" address="0x7ff8080295ed"/>
    <frame level="5" function="time2" offset="0" address="0x7ff807ebbdee"/>
    <frame level="6" function="TransactionListCtrl::OnEditTransaction(wxCommandEvent&amp;)" offset="0" address="0x1003c2057"/>
    <frame level="7" function="wxAppConsoleBase::HandleEvent(wxEvtHandler*, void (wxEvtHandler::*)(wxEvent&amp;), wxEvent&amp;) const" offset="0" address="0x100b9da33"/>
    <frame level="8" function="mmGUIApp::HandleEvent(wxEvtHandler*, void (wxEvtHandler::*)(wxEvent&amp;), wxEvent&amp;) const" offset="0" address="0x1003fc6f5"/>
    <frame level="9" function="wxAppConsoleBase::CallEventHandler(wxEvtHandler*, wxEventFunctor&amp;, wxEvent&amp;) const" offset="0" address="0x100b9dab4"/>
    <frame level="10" function="wxEvtHandler::ProcessEventIfMatchesId(wxEventTableEntryBase const&amp;, wxEvtHandler*, wxEvent&amp;)" offset="0" address="0x100ce5ba0"/>
    <frame level="11" function="wxEventHashTable::HandleEvent(wxEvent&amp;, wxEvtHandler*)" offset="0" address="0x100ce595b"/>
    <frame level="12" function="wxEvtHandler::TryHereOnly(wxEvent&amp;)" offset="0" address="0x100ce78aa"/>
    <frame level="13" function="wxEvtHandler::TryBeforeAndHere(wxEvent&amp;)" offset="0" address="0x100ce763f"/>
    <frame level="14" function="wxEvtHandler::ProcessEventLocally(wxEvent&amp;)" offset="0" address="0x100ce7671"/>
    <frame level="15" function="wxEvtHandler::ProcessEvent(wxEvent&amp;)" offset="0" address="0x100ce752e"/>
    <frame level="16" function="wxEvtHandler::ProcessPendingEvents()" offset="0" address="0x100ce7175"/>
    <frame level="17" function="wxAppConsoleBase::ProcessPendingEvents()" offset="0" address="0x100b9d56b"/>
    <frame level="18" function="wxCFEventLoop::CommonModeObserverCallBack(__CFRunLoopObserver*, int)" offset="0" address="0x100cba593"/>
    <frame level="19" function="wxCFEventLoop::OSXCommonModeObserverCallBack(__CFRunLoopObserver*, int, void*)" offset="0" address="0x100cba51c"/>
    <frame level="20" function="__CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__" offset="0" address="0x7ff8080d7584"/>
    <frame level="21" function="__CFRunLoopDoObservers" offset="0" address="0x7ff8080d74ab"/>
    <frame level="22" function="__CFRunLoopRun" offset="0" address="0x7ff8080d6a36"/>
    <frame level="23" function="CFRunLoopRunSpecific" offset="0" address="0x7ff8080d6071"/>
    <frame level="24" function="RunCurrentEventLoopInMode" offset="0" address="0x7ff811b3efcd"/>
    <frame level="25" function="ReceiveNextEventCommon" offset="0" address="0x7ff811b3ec14"/>
    <frame level="26" function="_BlockUntilNextEventMatchingListInModeWithFilter" offset="0" address="0x7ff811b3eb38"/>
    <frame level="27" function="_DPSNextEvent" offset="0" address="0x7ff80b1687a0"/>
    <frame level="28" function="-[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]" offset="0" address="0x7ff80b16764a"/>
    <frame level="29" function="-[NSApplication run]" offset="0" address="0x7ff80b159cb8"/>
    <frame level="30" function="wxGUIEventLoop::OSXDoRun()" offset="0" address="0x100947be6"/>
    <frame level="31" function="wxCFEventLoop::DoRun()" offset="0" address="0x100cbaf64"/>
    <frame level="32" function="wxEventLoopBase::Run()" offset="0" address="0x100bdc8e1"/>
    <frame level="33" function="wxAppConsoleBase::MainLoop()" offset="0" address="0x100b9c83a"/>
    <frame level="34" function="wxAppConsoleBase::OnRun()" offset="0" address="0x100b9c369"/>
    <frame level="35" function="wxAppBase::OnRun()" offset="0" address="0x1009a61e8"/>
    <frame level="36" function="wxApp::OnRun()" offset="0" address="0x1008e46a6"/>
    <frame level="37" function="wxEntry(int&amp;, wchar_t**)" offset="0" address="0x100c1f10d"/>
    <frame level="38" function="wxEntry(int&amp;, char**)" offset="0" address="0x100c1f2b6"/>
    <frame level="39" function="main" offset="0" address="0x1003fb100"/>
    <frame level="40" function="start" offset="0" address="0x7ff807ca241f"/>
  </stack>
</report>
```

**Comment by tactilis** [comment]
Looks to be a duplicate of this https://github.com/moneymanagerex/moneymanagerex/issues/5630

The erroneous transaction will be automatically removed from your database when you update to 1.6.4.

**Comment by fabianrinker** [comment]
@tactilis Thanks for linking that issue, looks similar.
I just don't remember that I have deleted a transaction and also in my backup files I can not find it. But probably it was deleted in the same session it was created.

The "Deleted Transaction" section is only shown when there are deleted transactions? (View option is checked, but I still don't see it)

**Comment by tactilis** [comment]
> The "Deleted Transaction" section is only shown when there are deleted transactions?

Correct.

And provided that this Option is set to greater than zero.

![image](https://github.com/moneymanagerex/moneymanagerex/assets/4060092/ab32dab7-f3b2-4345-a68c-8856c0de1e95)

**Comment by fabianrinker** [comment]
Okay, that is 30 days on my side but maybe I really deleted it from there already in the same session and just don't remember. Anyhow issue seems to be a duplicate so I close it now.

Thanks for your help @tactilis :)
